src/main/java/htsjdk/samtools/cram/build/CompressionHeaderFactory.java

```diff
@@ -479,56 +479,56 @@ private EncodingDetails buildEncodingForTag(final List<CramCompressionRecord> re
     private static class CompressionHeaderBuilder {
         private final CompressionHeader header;
 
-        CompressionHeaderBuilder(final boolean sorted) {
+        CompressionHeaderBuilder(final boolean coordinateSorted) {
             header = new CompressionHeader();
             header.externalIds = new ArrayList<>();
             header.tMap = new TreeMap<>();
 
             header.encodingMap = new TreeMap<>();
-            header.APDelta = sorted;
+            header.APDelta = coordinateSorted;
         }
 
         CompressionHeader getHeader() {
             return header;
         }
 
-        void addExternalEncoding(final EncodingKey encodingKey, final EncodingParams params,
+        void addExternalEncoding(final DataSeries dataSeries, final EncodingParams params,
                                  final ExternalCompressor compressor) {
-            header.externalIds.add(encodingKey.ordinal());
-            header.externalCompressors.put(encodingKey.ordinal(), compressor);
-            header.encodingMap.put(encodingKey, params);
+            header.externalIds.add(dataSeries.ordinal());
```

Comment:
The ordinal values function as external block content IDs - I'll make that more explicit.

Oddly, the spec says it can't be zero - but the first value (formerly `BF_bitflags`) would have had a zero here.  Assuming this isn't enforced.  Also, htslib uses a different ordering and has a comment explicitly stating that it's specific to the implementation.

Two-letter data series names are stored in the Compression Header.  I'll make that access more explicit as well.